### ISSUE TITLE
2.0 Lost name/service information of the keys

### ISSUE THREAD

**Issue by vacavaca** [freeotp/freeotp-android#291]
1. Seems like a critical bug: on 2.0 the UI doesn't show names for most of the keys, so I can not determine which OTP belongs to which service. Having dozens of keys this makes the app unusable.

2. Is there a way to switch back to the previous version for now or export raw secrets from the app?

![Photo on 09 01 2023 at 21 36](https://user-images.githubusercontent.com/19508396/211384025-69bb0ef1-208d-4403-aea0-45b7ff276b5c.jpg)
 
(Sorry for the quality of the pic, the app doesn't allow to make screenshots)

**Comment by vacavaca** [comment]
In the backup XML I see empty "issuerExt" fields on the problematic keys, but "issuerInt" still contains all the relevant names. I believe the issue is that app does not show them anymore in the UI.

**Comment by justin-stephenson** [comment]
Can you try pressing the token icon, then press the Pencil button to edit the Issuer/Account fields? I can post a quick video if needed.

I'm not sure what caused the account information to be dropped, this is the first report I've seen of this problem. As you can see at the bottom we display the Account and Issuer fields as listed here: https://freeotp.github.io/qrcode.html

**Comment by vacavaca** [comment]
> Can you try pressing the token icon, then press the Pencil button to edit the Issuer/Account fields? I can post a quick video if needed.

Opened the edit dialog, but still only seeing my email (account) linked with the token but not any issuer information. Two text inputs there, first filled with the account, and the second one is empty.

**Comment by justin-stephenson** [comment]
If you see the correct names in then can you add the 'IssuerInt' manually using the edit token (pencil) functionality?

Please post the XML contents of the token with missing information.

**Comment by vacavaca** [comment]
> can you add the 'IssuerInt' manually

Yep, I think I could match the tokens in the app with the ones in the XML.

Here is the part of XML (redacted to remove sensitive parts):
`{"algo":"SHA1","counter":0,"digits":6,"issuerExt":"","issuerInt":"<redacted, valid issuer name here>","label":"<redacted, my email here>@gmail.com-1585824713","period":30,"type":"TOTP"}`

BTW, is there a way to decrypt the backup or export secrets some other way? I think I need to do a backup to my desktop to prevent this in the future.

**Comment by justin-stephenson** [comment]
> Here is the part of XML (redacted to remove sensitive parts): `{"algo":"SHA1","counter":0,"digits":6,"issuerExt":"","issuerInt":"<redacted, valid issuer name here>","label":"<redacted, my email here>@gmail.com-1585824713","period":30,"type":"TOTP"}`
> 

Thank you, it appears we don't handle `issuerInt` properly. This looks to be a legacy parameter which is now replaced with `issuer`  https://github.com/freeotp/freeotp-android/blob/master/URI.md



> BTW, is there a way to decrypt the backup or export secrets some other way? I think I need to do a backup to my desktop to prevent this in the future.

Please see https://github.com/freeotp/freeotp-android/blob/master/BACKUP.md for backup methods.
